### PR TITLE
chore: launch code review

### PR DESCRIPTION
**Pull request by jak-pan**

### CODE REVIEW

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -288,6 +289,7 @@ impl pallet_timestamp::Config for Runtime {
 }
 
 parameter_types! {
+	// TODO: Shouldn't this be at least 1?
```
Comment (jak-pan) [inline runtime/src/lib.rs:292]: @enthusiastmartin maybe 1000 like duster?

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -353,6 +355,8 @@ parameter_type_with_key! {
 		// Dusting is handled by duster pallet.
 		// However, to make sure that account is reaped/killed and storage updated, ED must be > 0
 		// On ED = 0 - accounts are never reaped.
+
+		// We might want to set this to the same thing as duster e.g. 1000 for everything as a base
```
Comment (jak-pan) [inline runtime/src/lib.rs:359]: @enthusiastmartin

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -429,6 +434,7 @@ impl pallet_exchange::Config for Runtime {
 }
 
 parameter_types! {
+	// Is this ok?
```
Comment (jak-pan) [inline runtime/src/lib.rs:437]: @enthusiastmartin

Comment (enthusiastmartin) [reply runtime/src/lib.rs:438]: the value is ok. I am not sure what the reward amount should be, hence 0 for now.

We can also consider doiing this #122

File: runtime/src/lib.rs (hunk 2 above)
Comment (enthusiastmartin) [reply runtime/src/lib.rs:359]: Yes, we can set all ED to 1000.  

The only is that there is not dust handler - so the dust will be lost at the moment.

File: runtime/src/lib.rs (hunk 1 above)
Comment (enthusiastmartin) [reply runtime/src/lib.rs:292]: Yes, we can set all ED to 1000.  

The only is that there is not dust handler - so the dust will be lost at the moment.

Comment (jak-pan) [reply runtime/src/lib.rs:292]: Send to treasury?

File: runtime/src/lib.rs (hunk 3 above)
Comment (jak-pan) [reply runtime/src/lib.rs:438]: The reward should be bigger/same as the amount paid for the tx but smaller than the dust amount.

File: primitives/src/traits.rs [hunk 4]
```diff
@@ -34,7 +34,7 @@ pub struct AMMTransfer<AccountId, AssetId, AssetPair, Balance> {
 
 /// Traits for handling AMM Pool trades.
 pub trait AMM<AccountId, AssetId, AssetPair, Amount> {
-	/// Check if both assets exist in a pool.
+	/// Check if both assets exist in a pool. //TODO: not sure if AssetPair is right here since we don't have asset_in or asset_out when not trading.
```
Comment (enthusiastmartin) [inline primitives/src/traits.rs:37]: Yeah, i was actually thinking that this AMM trait should be more generic to cover various pool implementations.

I have some ideas - but do we want to refactor now ?

File: runtime/src/currency.rs [hunk 5]
```diff
@@ -1,3 +1,5 @@
+// TODO: What is the purpose of this?
```
Comment (enthusiastmartin) [inline runtime/src/currency.rs:1]: No longer used. this was used in original implementation of xcm support.

I will remove it.

File: primitives/src/traits.rs (hunk 4 above)
Comment (jak-pan) [reply primitives/src/traits.rs:37]: Probably not. But let's start working on it?

Comment (jak-pan) [reply primitives/src/traits.rs:37]: After launch of crowdloan

File: runtime/src/lib.rs (hunk 2 above)
Comment (jak-pan) [reply runtime/src/lib.rs:359]: Let's actually do 1_000_000 everywhere. It's still 6 decimals so it's like 4 cents if you are dusting bitcoin. Anything else mostly is peanuts. We should do it dynamic tho asap.

File: primitives/src/traits.rs (hunk 4 above)
Comment (enthusiastmartin) [reply primitives/src/traits.rs:37]: #145

File: runtime/src/lib.rs (hunk 2 above)
Comment (jak-pan) [reply runtime/src/lib.rs:357]: Maybe for now add default existential deposit to assetRegistry + set default deposits for generated assets to primitives/MIN_POOL_LIQUIDITY

File: pallets/asset-registry/src/tests.rs [hunk 6]
```diff
@@ -206,3 +208,54 @@ fn set_metadata_works() {
 			);
 		});
 }
+
+#[test]
+fn update_asset() {
```
Comment (enthusiastmartin) [inline pallets/asset-registry/src/tests.rs:213]: These tests actually dont really test the the storage was updated. They only check the result of the update call.

Comment (green-jay) [reply pallets/asset-registry/src/tests.rs:213]: added

File: primitives/src/traits.rs (hunk 4 above)
Comment (jak-pan) [inline primitives/src/traits.rs:37]:
```suggestion
	/// Check if both assets exist in a pool.
```

File: pallets/lbp/src/lib.rs [hunk 7]
```diff
@@ -857,10 +865,17 @@ impl<T: Config> Pallet<T> {
 			let amount_in_with_fee = token_amount_in.checked_add(transfer_fee).ok_or(Error::<T>::Overflow)?;
 
 			ensure!(limit >= amount_in_with_fee, Error::<T>::AssetBalanceLimitExceeded);
-
 			(token_amount_in, amount, assets.asset_in, transfer_fee)
 		};
 
+		// TODO: check if user has enough tokens in?
```
Comment (jak-pan) [inline pallets/lbp/src/lib.rs:871]:
```suggestion
```

File: runtime/src/lib.rs [hunk 8]
```diff
@@ -429,30 +428,19 @@ impl pallet_exchange::Config for Runtime {
...
```
Comment (jak-pan) [inline runtime/src/lib.rs:431]:
```suggestion
```

**Review by jak-pan — COMMENTED** [review]
LGTM

**Review by lumir-mrkva — APPROVED** [review]

**Pull request merged**